packages/taquito/src/taquito.ts

```diff
@@ -336,6 +343,23 @@ export class TezosToolkit {
     }
   }
 
+  /**
+   * @description Sets injector provider on the Tezos Taquito instance
+   *
+   * @param options Injector to use to interact with the Tezos network by default RpcInjector
+   *
+   */
+  setInjectorProvider(injectorProvider?: SetProviderOptions['injectorProvider']) {
+    if (!this._options.injectorProvider && typeof injectorProvider === 'undefined') {
+      const p = new RpcInjector(this._context);
+      this._context.injector = p;
+      this._options.injectorProvider = p;
+    } else if (typeof injectorProvider !== 'undefined') {
+      this._context.injector = injectorProvider;
+      this._options.injectorProvider = injectorProvider;
+    }
+  }
```

Comment:
What happens if `this._options.injectorProvider` is truthy? Neither of the `if` statements will be executed, and the code does nothing without giving feedback to the calling code. So the user thinks they have set the injectorProvider, but they have not.